File: spacexinterop.client/src/app/shared/enums/api/ResultStatusEnum.ts
export enum ResultStatusEnum {
  //#region Basic
  Default = 'Default',
  Success = 'Success',
  Failure = 'Failure',
  NotFound = 'NotFound',
  ValidationFailed = 'ValidationFailed',
  Exception = 'Exception',
  InvalidType = 'InvalidType',
  //#endregion

  //#region Auth
  Unauthorized = 'Unauthorized',
  EmailAlreadyExists = 'EmailAlreadyExists',
  //#endregion
}

File: spacexinterop.api/_Common/Domain/Data/Result/ResultStatusEnum.cs
using spacexinterop.api._Common.Utility.Attributes;

namespace spacexinterop.api._Common.Domain.Data.Result;

public enum ResultStatusEnum : byte
{
    #region Basic
    Default,

    [ResultStatusConnotation(IsPositive = true)]
    Success,

    Failure,

    NotFound,
    ValidationFailed,
    Exception,
    InvalidRequest,
    #endregion

    #region Auth
    Unauthorized,
    EmailAlreadyExists
    #endregion
}